src/js/helpers.js

```diff
@@ -218,16 +223,7 @@ function midi_note_number_to_name(input) {
 
 // calculate the sum of the values in a given array given a stopping index
 function sum_array(array, index) {
-  var sum = 0;
-
-  if (array.length <= index)
-    index = array.length - 1;
-
-  for (var i = 0; i < index; i++) {
-    sum += array[i];
-  }
-
-  return sum;
+  return array.reduce((sum, x) => sum + x, 0)
```

Comment:
Watch out, you are ignoring the `index` parameter, which tells the function where to stop adding up values. Add something like `array.slice(0, index)` to utilize index.